sling/myelin/gradient.cc

```diff
@@ -122,6 +122,8 @@ Flow::Function *Gradient(Flow *flow,
   Gradients g(flow, func, vars);
   for (int i = ops.size() - 1; i >= 0; --i) {
     Flow::Operation *op = ops[i];
+    if (op->is(Flow::Operation::NOGRADIENT)) continue;
```

Comment:
Yes, that is true. As the code is now, you have to mark all downstream ops as NOGRADIENT manually for now. It would require more analysis to skip all the ops that are not needed for the gradient computation, so I will postpone this to a future PR. Right now it is only being used for implementing "static" word embedding in the lexical encoder trainer.